### ISSUE TITLE
uploading contact from cardbook to radicale very slow and error prone

### ISSUE THREAD

**Issue by hakova** [Kozea/Radicale#856]
Hi all,

First of all thank you for developing this wonderful tool. I have been using it for almost a year now. I originally installed it on a FreeBSD 11.1 server, but the whole filesystem corrupted while trying to upgrade FreeBSD, so I recently set it up again on a Manjaro linux VM. I did not have any issue in uploading my contacts (1048 total) to Radicale using Cardbook add-on to Thunderbird with the initial server. However, with the new setup, every time  Thunderbird tries to synchronize with Radicale, it can only upload 3-5 contacts. Its log reports several errors in creating the remaining contacts on the server (unable to create contact so and so, etc.). It has been 3 days since I had radicale up and running, and thunderbird is attempting to synchronize every 10 minutes around the clock, I currently have 203 contacts uploaded. To be fair, the uploaded contacts seem intact with their info and pictures, etc.

Another problem is the web interface gives "Error: 0"  at least half the time I try to login with my user name / password. This has never happened before with the previous set up. Of note, I always used https with a LetsEncrypt certificate.

One difference I have with the new set up is now I have 3 total users, whereas it used to be only 1 in the former set up. Only 1 out of 3 users currently has collections (contacts and calendar) in the server though.

While troubleshooting, I disabled reverse-DNS lookup feature, which didn't seem to help. journalctl occasionally reports SSL handshake issues, and warning message that a line failed to be parsed during a bad PUT request involving a Vcard.

I appreciate any troubleshooting suggestions, pointers or recommendations. Thank you for tolerating my novice status.

**Comment by hakova** [comment]
This is just a follow up to my previous post. I built a FreeBSD VM and set up radicale on it just to test if it was the VM versus actual server that made the difference or the OS instead. Radicale on FreeBSD VM worked like a charm. Using the same method (uploading VCards via Cardbook), it took less tahn 30 seconds to transfer 1039 contacts to the server. Nine of them had some parsing problem, which I didn't have a chance to troubleshoot yet. Therefore, I concluded that radicale doesn't work as robust as it does on Manjaro Linux as compared to FreeBSD. I am not sure what exactly creates this difference, but this may hopefully give somebody a lead in the future, who is interested in troubleshooting this.
